system-tests/e2e-transfer-test/runner/src/test/java/org/eclipse/edc/aws/test/e2e/S3DataPlaneProvisionEndToEndTest.java

```diff
@@ -0,0 +1,352 @@
+                .userName(sourceUser)
+                .build());
+        sourceAccessKeyId = sourceCredentials.accessKey().accessKeyId();
+        sourceSecretAccessKey = sourceCredentials.accessKey().secretAccessKey();
+
+        System.setProperty(SYSTEM_PROPERTY_AWS_ACCESS_KEY_ID, sourceAccessKeyId);
+        System.setProperty(SYSTEM_PROPERTY_AWS_SECRET_ACCESS_KEY, sourceSecretAccessKey);
+
+        // consumer
+        iamClient.createPolicy(CreatePolicyRequest.builder()
+                .policyName(destinationUserPolicyName)
+                .policyDocument(destinationUserPolicy())
+                .build());
+        iamClient.createUser(CreateUserRequest.builder()
+                .userName(destinationUser)
+                .build());
+        iamClient.attachUserPolicy(AttachUserPolicyRequest.builder()
+                .userName(destinationUser)
+                .policyArn("arn:aws:iam::000000000000:policy/" + destinationUserPolicyName)
+                .build());
+        var destinationCredentials = iamClient.createAccessKey(CreateAccessKeyRequest.builder()
+                .userName(destinationUser)
+                .build());
+        destinationAccessKeyId = destinationCredentials.accessKey().accessKeyId();
+        destinationSecretAccessKey = destinationCredentials.accessKey().secretAccessKey();
+    }
+
+    @AfterEach
+    void tearDown() {
+        LOCALSTACK_CONTAINER.stop();
+
+        System.clearProperty(SYSTEM_PROPERTY_AWS_ACCESS_KEY_ID);
+        System.clearProperty(SYSTEM_PROPERTY_AWS_SECRET_ACCESS_KEY);
+    }
+
+    @Test
+    void provision_createsBucketAndRole_withCredentialsFromDataDestination() {
+        createConsumerSecret("s3-credentials", awsSecretToken(destinationAccessKeyId, destinationSecretAccessKey));
+
+        createAsset(LOCALSTACK_CONTAINER.getEndpoint().toString());
+        createPolicy();
+        createContractDefinition();
+
+        var negotiationId = initiateNegotiation();
+        waitForNegotiationState(negotiationId, FINALIZED);
+        var agreementId = getAgreementId(negotiationId);
+
+        var transferId = initiateTransfer(
+                agreementId,
+                LOCALSTACK_CONTAINER.getEndpoint().toString(),
+                LOCALSTACK_CONTAINER.getAccessKey(),
+                LOCALSTACK_CONTAINER.getSecretKey());
+        waitForTransferState(transferId, COMPLETED);
+
+        var destinationFileContent = readS3DestinationObject();
+        assertThat(destinationFileContent).isEqualTo(fileContent);
+
+        waitForProviderTransferState(transferId, DEPROVISIONED);
+
+        var s3Client = getS3Client();
+        assertThatThrownBy(() -> s3Client.getBucketPolicy(GetBucketPolicyRequest.builder()
+                .bucket(destinationBucket)
+                .build()))
+                .isInstanceOfSatisfying(S3Exception.class, ex -> assertThat(ex.awsErrorDetails().errorCode().equals("NoSuchBucketPolicy")));
```

Comment:
do we needs the bucket policy check here? The consumer provisioner does not add or modify the bucket policy